ascend-sdk/src/main/java/ai/evolv/ascend/android/Allocator.java

```diff
@@ -36,19 +72,112 @@ HttpUrl createAllocationsUrl() {
                 .build();
     }
 
-    synchronized JsonArray fetchAllocations() throws AscendAllocationException {
-        JsonArray allocations;
+    Future<JsonArray> fetchAllocations() {
+        OkHttpClient client = new OkHttpClient.Builder()
+                .callTimeout(config.getTimeout(), TimeUnit.MILLISECONDS)
+                .retryOnConnectionFailure(false)
+                .build();
+
+        final Request request = new Request.Builder()
+                .url(createAllocationsUrl())
+                .build();
+
+        final SettableFuture<JsonArray> responseFuture = SettableFuture.create();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override public void onFailure(Call call, IOException e) {
+                resolveAllocationFailure(responseFuture);
+            }
+
+            @Override public void onResponse(Call call, Response response) {
```

Comment:
I find putting annotations on their own line to be more readable. Totally subjective though.

Reply:
Yah, lets be consistent